backend/src/main/java/org/cryptomator/hub/keycloak/KeycloakAdminService.java

```diff
@@ -0,0 +1,444 @@
+
+		UserResource userResource = realm.users().get(userId);
+
+		if (pictureUrl != null && !pictureUrl.isBlank()) {
+			UserRepresentation createdUser = userResource.toRepresentation();
+			createdUser.setAttributes(Map.of("picture", List.of(pictureUrl)));
+			userResource.update(createdUser);
+		}
+
+		if (groupIds != null && !groupIds.isEmpty()) {
+			for (String groupId : groupIds) {
+				try {
+					userResource.joinGroup(groupId);
+				} catch (Exception e) {
+					LOG.warn("Failed to add user {} to group {}", userId, groupId, e);
+				}
+			}
+		}
+
+		syncUser(userId);
+
+		return realm.users().get(userId).toRepresentation();
+	}
+
+	public UserRepresentation getUser(String userId) {
+		RealmResource realm = keycloak.realm(keycloakRealm);
+		try {
+			return realm.users().get(userId).toRepresentation();
+		} catch (jakarta.ws.rs.NotFoundException e) {
+			throw new NotFoundException("User not found: " + userId);
+		}
+	}
+
+	public UserRepresentation updateUser(String userId, String firstName, String lastName, String password, String pictureUrl) {
+		if (isUserReadOnly(userId)) {
+			throw new ForbiddenException("User has a federated identity and cannot be modified");
+		}
+
+		RealmResource realm = keycloak.realm(keycloakRealm);
+		UserResource userResource;
+		try {
+			userResource = realm.users().get(userId);
+		} catch (jakarta.ws.rs.NotFoundException e) {
+			throw new NotFoundException("User not found: " + userId);
+		}
+
+		UserRepresentation user = userResource.toRepresentation();
+
+		if (firstName != null && !firstName.isBlank()) {
+			user.setFirstName(firstName);
+		}
+		if (lastName != null && !lastName.isBlank()) {
+			user.setLastName(lastName);
+		}
+		if (pictureUrl != null) {
+			Map<String, List<String>> attrs = user.getAttributes();
+			if (attrs == null) {
+				attrs = new java.util.HashMap<>();
+			} else {
+				attrs = new java.util.HashMap<>(attrs);
+			}
+			if (pictureUrl.isBlank()) {
+				attrs.remove("picture");
+			} else {
+				attrs.put("picture", List.of(pictureUrl));
+			}
+			user.setAttributes(attrs);
+		}
+
+		userResource.update(user);
+
+		if (password != null && !password.isBlank()) {
+			CredentialRepresentation credential = new CredentialRepresentation();
+			credential.setType(CredentialRepresentation.PASSWORD);
+			credential.setValue(password);
+			credential.setTemporary(false);
+			userResource.resetPassword(credential);
+		}
+
+		syncUser(userId);
+		return userResource.toRepresentation();
+	}
+
+	@Transactional
+	public void deleteUser(String userId) {
+		if (isUserReadOnly(userId)) {
+			throw new ForbiddenException("User has a federated identity and cannot be deleted");
+		}
+
+		RealmResource realm = keycloak.realm(keycloakRealm);
+		try {
+			realm.users().delete(userId);
+		} catch (jakarta.ws.rs.NotFoundException e) {
+			throw new NotFoundException("User not found: " + userId);
+		}
+
+		User user = userRepo.findById(userId);
+		if (user != null) {
+			userRepo.delete(user);
+		}
```

Comment:
Why do we delete here the user in the database directly? This should be the job of the syncer.